Composer/packages/adaptive-form/src/utils/__tests__/getOrderedProperties.test.ts

```diff
@@ -78,4 +78,36 @@ describe('getOrderedProperties', () => {
       'multiple wildcards'
     );
   });
+
+  it('throws an exception if additional field name already exists in the schema', () => {
+    const uiOptions = { additionalFields: [{ name: 'one', field: 'field' }] };
+    expect(() => getOrderedProperties(schema, uiOptions, data)).toThrow(
+      'additional field name already exists in schema'
+    );
+  });
+
+  it('includes additional fields', () => {
+    const uiOptions = { additionalFields: [{ name: 'additionalField', field: 'field' }] };
+    expect(getOrderedProperties(schema, uiOptions, data)).toEqual(
+      expect.arrayContaining([{ name: 'additionalField', field: 'field' }])
+    );
+  });
+
+  it('can include additional fields in the order', () => {
+    const uiOptions = {
+      additionalFields: [{ name: 'additionalField', field: 'field' }],
+      order: ['one', 'two', 'additionalField', '*'],
+    };
+
+    expect(getOrderedProperties(schema, uiOptions, data)).toEqual([
+      'one',
+      'two',
+      { name: 'additionalField', field: 'field' },
```

Comment:
Ah. This answers my previous question.

What if instead this just returns the name of the field and the field resolution logic would know to check for additional fields? That would isolate the knowledge of additional fields a little more to the ui options utility functions.

Reply:
We could just return the name, but if we don't add the additional field to the schema, we would have to map it to the additional field object at some point. `getOrderedProperties` just seemed like a convenient place to handle that.